ui/app/adapters/generated-item-list.js

```diff
@@ -6,42 +6,111 @@
 import ApplicationAdapter from './application';
 import { task } from 'ember-concurrency';
 import { service } from '@ember/service';
+import { sanitizePath } from 'core/utils/sanitize-path';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { tracked } from '@glimmer/tracking';
 
-export default ApplicationAdapter.extend({
-  store: service(),
-  namespace: 'v1',
-  urlForItem() {},
-  dynamicApiPath: '',
+export default class GeneratedItemListAdapter extends ApplicationAdapter {
+  @service store;
+  namespace = 'v1';
 
-  getDynamicApiPath: task(function* (id) {
-    // TODO: remove yield at some point.
-    const result = yield this.store.peekRecord('auth-method', id);
-    this.dynamicApiPath = result.apiPath;
-    return;
-  }),
+  // these items are set within getNewAdapter in path-help service
+  @tracked apiPath = '';
+  paths = {};
 
-  fetchByQuery: task(function* (store, query, isList) {
+  // These are the paths used for the adapter actions
+  get getPath() {
+    return this.paths.getPath || '';
+  }
+  get createPath() {
+    return this.paths.createPath || '';
+  }
+  get deletePath() {
+    return this.paths.deletePath || '';
+  }
+
+  getDynamicApiPath(id) {
+    const result = this.store.peekRecord('auth-method', id);
+    this.apiPath = result.apiPath;
+    return result.apiPath;
+  }
+
+  fetchByQuery = task(async (store, query, isList) => {
     const { id } = query;
-    const data = {};
+    const payload = {};
     if (isList) {
-      data.list = true;
-      yield this.getDynamicApiPath.perform(id);
+      payload.list = true;
     }
+    const path = isList ? this.getDynamicApiPath(id) : '';
 
-    return this.ajax(this.urlForItem(id, isList, this.dynamicApiPath), 'GET', { data }).then((resp) => {
-      const data = {
-        id,
-        method: id,
-      };
-      return { ...resp, ...data };
-    });
-  }),
+    const resp = await this.ajax(this.urlForItem(id, isList, path), 'GET', { data: payload });
+    const data = {
+      id,
+      method: id,
+    };
+    return { ...resp, ...data };
+  });
 
   query(store, type, query) {
     return this.fetchByQuery.perform(store, query, true);
-  },
+  }
 
   queryRecord(store, type, query) {
     return this.fetchByQuery.perform(store, query);
-  },
-});
+  }
+
+  urlForItem(id, isList, dynamicApiPath) {
+    const itemType = sanitizePath(this.getPath);
+    let url;
+    id = encodePath(id);
```

Comment:
For my own understanding, because we're encoding the id here, i would expect it could be something like `/test?`. Is that correct? ID to me indicates a string that shouldn't need encoding, but then again we manual change ids to be concatenations of strange things.

Reply:
I believe this can be anything that an auth item (such as userpass username) can be. In that case, things like spaces or special characters need to be encoded, which is why we do this here.